Realm/RLMSyncSubscription.h

```diff
@@ -0,0 +1,336 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+@class RLMObjectId;
+
+#pragma mark - Subscription States
+
+typedef NS_ENUM(NSUInteger, RLMSyncSubscriptionState) {
+    /// The subscription is complete and the server has sent all the data that matched the subscription
+    /// queries at the time the subscription set was updated. The server is now in a steady-state
+    /// synchronization mode where it will stream update as they come.
+    RLMSyncSubscriptionStateComplete,
+    /// The subscription encountered an error and synchronization is paused for this Realm. You can
+    /// find the error calling error in the subscription set to get a description of the error. You can
+    /// still use the current subscription set to write a subscription.
+    RLMSyncSubscriptionStateError,
+    /// The subscription is persisted locally but not yet processed by the server, which means
+    /// the server hasn't yet returned all the data that matched the updated subscription queries.
+    RLMSyncSubscriptionStatePending,
+    /// The subscription set has been super-ceded by an updated one, this typically means that
+    /// someone is trying to write a subscription on a different instance of the subscription set.
+    /// You should not use a super-ceded subscription set and instead obtain a new instance of
+    /// the subscription set to write a subscription.
+    RLMSyncSubscriptionStateSuperceded
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RLMSyncSubscription : NSObject
+
+/// Identifier of the subscription.
+@property (nonatomic, readonly) RLMObjectId *identifier;
+
+/// Name of the subscription. If not specified it will return nil.
+@property (nonatomic, readonly, nullable) NSString *name;
+
+/// When the subscription was created. Recorded automatically.
+@property (nonatomic, readonly) NSDate *createdAt;
+
+/// When the subscription was last updated. Recorded automatically.
+@property (nonatomic, readonly) NSDate *updatedAt;
+
+/**
+ Updates a Flexible Sync's subscription query with an allowed query which will be used to bootstrap data
+ from the server when committed.
+
+ @warning This method may only be called during a write subscription block.
+
+ @param objectClassName The class name for the model class to be queried.
+ @param predicateFormat A predicate format string, optionally followed by a variable number of arguments.
+ */
+- (void)updateSubscriptionWithClassName:(NSString *)objectClassName
```

Comment:
I'm confused by this API. This seems to suggest that a Subscription has a query on one or more classes? Or exactly one class at a time, and this changes what class the Subscription is for?

Reply:
cc @dianaafanador3

Reply:
A subscription only have one associated class at a time, it doesn't mean that this class cannot be changed. We can update a subscription with a query associated to a different class that the previous one. For example you can have a subscription named `initial_query` which can have an initial query `Project.id == 'XXXXXX'` which can be updated to `User.id == 'XXXXXX'`, which is using a different object class for the same subscription.